test/PushNotificationsTests/APITests.cpp

```diff
@@ -0,0 +1,106 @@
+
+namespace Test::PushNotifications
+{
+    class APITests
+    {
+    private:
+        wil::unique_event m_failed;
+
+        const std::wstring c_testDataFileName = L"testfile" + c_testFileExtension;
+        const std::wstring c_testDataFileName_Packaged = L"testfile" + c_testFileExtension_Packaged;
+        const std::wstring c_testPackageFile = g_deploymentDir + L"AppLifecycleTestPackage.msixbundle";
+        const std::wstring c_testPackageCertFile = g_deploymentDir + L"AppLifecycleTestPackage.cer";
+        const std::wstring c_testPackageFullName = L"AppLifecycleTestPackage_1.0.0.0_x64__ph1m9x8skttmg";
+
+    public:
+        BEGIN_TEST_CLASS(APITests)
+            TEST_CLASS_PROPERTY(L"IsolationLevel", L"Method")
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunFixtureAs:Class", L"RestrictedUser")
+            END_TEST_CLASS()
+
+            TEST_CLASS_SETUP(ClassInit)
+        {
+            ::Test::Bootstrap::SetupPackages();
+
+            // Write out some test content.
+            WriteContentFile(c_testDataFileName);
+            WriteContentFile(c_testDataFileName_Packaged);
+
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            // Swallow errors in cleanup.
+            try
+            {
+                DeleteContentFile(c_testDataFileName_Packaged);
+                DeleteContentFile(c_testDataFileName);
+                //UninstallPackage(c_testPackageFullName);
+            }
+            catch (const std::exception&)
+            {
+            }
+            catch (const winrt::hresult_error&)
+            {
+            }
+
+            ::Test::Bootstrap::CleanupPackages();
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_ProjectReunionFramework());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+
+            m_failed = CreateTestEvent(c_testFailureEventName);
+
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_ProjectReunionFramework());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_DynamicDependencyLifetimeManager());
+
+            return true;
+        }
+
+        TEST_METHOD(GetActivatedEventArgsIsNull_UAP)
+        {
+            BEGIN_TEST_METHOD_PROPERTIES()
+                TEST_METHOD_PROPERTY(L"RunAs", L"UAP")
+                TEST_METHOD_PROPERTY(L"UAP:AppxManifest", L"PushNotifications-AppxManifest.xml")
+                END_TEST_METHOD_PROPERTIES();
+
+            VERIFY_IS_TRUE(true);
+            // (winrt::Microsoft::ApplicationModel::Activation::AppLifecycle::GetActivatedEventArgs());
```

Comment:
Remove comment here? It doesnt seem relevant to the checkin.

Reply:
The line is supposed to be VERIFY_IS_NULL(winrt::Microsoft::ApplicationModel::Activation::AppLifecycle::GetActivatedEventArgs()); to test the unit test ran as packaged. Removing the VERIFY_IS_TRUE and replacing.

Reply:
Curious, but where is the WinMain implementation for this? We are relying on WinMain Activation events for the packaged MSIX feature set. So need top make sure that this works with @danielayala94 s changes

Reply:
I see in AppLifecycle tests that it launches its own Win32 app to verify activation type and signal events accordingly, and the TAEF tests wait for that event signaling. @hulumane is this the pattern that we are looking for?